cypress/integration/admin_portal/helpers/helper_functions.js

```diff
@@ -19,6 +21,43 @@ class HelperFunctions {
     return coupons.fetchCouponReport(couponId).then((response) => {
     })
   }
+
+  verifymodalWindow(win, couponName, remainingRedemptions){
+    const fileName = 'valid_emails.csv'
+    const fileType = 'application/csv'
+    const fileSelector = '#csv-email-addresses'
+    const modelWindowLabelsAndText = {
+      modelLabel: 'Code Assignment',
+      unassignedCodes: 'Unassigned Codes: ',
+      selectedCodes: 'Selected Codes: ',
+      helpEmailText: 'To add more than one user, enter one email address per line.',
+      uploadCSVLabel: 'Upload Email Addresses',
+      csvHelpText: 'The file must be a CSV containing a single column of email addresses.',
+      maxNumbersEmailError: 'You have 2 codes selected, but you entered 3 emails. Please try again.',
+      csvEmailValidationError: 'Unable to assign codesYou uploaded a CSV and manually entered email addresses. Please only use one of these fields.'
+    }
+    cy.wrap(win).find('.modal-title>span').should('have.text', couponName)
+      cy.wrap(win).find('.modal-title>small').should('have.text', modelWindowLabelsAndText.modelLabel)
+      cy.wrap(win).find('.modal-body>div>p:nth-child(1)').should('have.text', modelWindowLabelsAndText.unassignedCodes+remainingRedemptions)
+      cy.wrap(win).find('.modal-body>div>p:nth-child(2)').should('have.text', modelWindowLabelsAndText.selectedCodes+(remainingRedemptions-1))
+      cy.wrap(win).find('#description-email-addresses').should('have.text', modelWindowLabelsAndText.helpEmailText)
+      cy.wrap(win).find('#description-email-addresses').should('have.text', modelWindowLabelsAndText.helpEmailText)
+      cy.wrap(win).find('.file-input>label').should('have.text', modelWindowLabelsAndText.uploadCSVLabel)
+      cy.wrap(win).find('#description-csv-email-addresses').should('have.text', modelWindowLabelsAndText.csvHelpText)
+      cy.wrap(win).find('textarea[name="email-addresses"]').type('cypress1@edx.org').type('{enter}').type('cypress2@edx.com').type('{enter}').type('cypress3@edx.org')
+      cy.upload_file(fileName, fileType, fileSelector).then(() => {
+        cy.wrap(win).find('.modal-footer .btn:nth-of-type(1)').click()
+        cy.wrap(win).find('.alert-dialog .message').should('have.text',modelWindowLabelsAndText.csvEmailValidationError)
+        cy.wrap(win).find('.remove-file-btn').click()
+        cy.wrap(win).find('.modal-footer .btn:nth-of-type(1)').click()
+        cy.wrap(win).find('#validation-email-addresses').should('have.text',modelWindowLabelsAndText.maxNumbersEmailError)
+        cy.wrap(win).find('textarea[name="email-addresses"]').type('{selectall}').type('cypress1@edx.org').type('{enter}').type('cypress2@edx.com')
+        cy.wrap(win).find('.modal-footer .btn:nth-of-type(1)').click()
```

Comment:
seems like we are doing a whole lot of assertions in helper function, can you explain why?

Reply:
I've moved this piece of code from helper function to tests.